lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -15,17 +15,45 @@ defmodule Mix.Tasks.Xref do
   The following commands are available:
 
     * `warnings` - prints warnings for violated cross reference checks
+
     * `unreachable` - prints all unreachable "file:line: module.function/arity" entries
+
     * `callers CALLEE` - prints all references of `CALLEE`, which can be one of: `Module`,
       `Module.function`, or `Module.function/arity`
 
+    * `graph` - prints the module reference graph. By default, an edge from `A` to `B` indicates
+      that `A` depends on `B`
+
   ## Command line options
 
     * `--no-compile` - do not compile even if files require compilation
+
     * `--no-deps-check` - do not check dependencies
+
     * `--no-archives-check` - do not check archives
+
     * `--no-elixir-version-check` - do not check the Elixir version from mix.exs
 
+    * `--exclude` - for the `graph` command, modules to exclude. Use `Module` for elixir modules,
```

Comment:
I think we can have those options per mode now. We can document a separate section per mode and have a section with the shared options. Implementation wise, it just means we need to call OptionParser multiple times.